### PR TITLE
Next.js PR 9: Add component props table and more package info

### PR DESCRIPTION
**Pull request by danoc**
This uses `react-docgen` and `doctrine` to parse the React component files and create the props table. The stable version of `react-docgen` doesn't have types, so I added some. Also, I moved the MDX components (`H1`, `P`, `Table`, etc.) from `mdx/mdx.tsx` to `mdx/components.tsx` to avoid a cyclical import issue.

### CODE REVIEW

File: next/components/thumbprint-components/props-table/prop-type.jsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import React from 'react';
```
Comment (lavelle) [inline next/components/thumbprint-components/props-table/prop-type.jsx:1]: is this file copied from the old gatsby docs? do we still need to support proptypes if all components are written in TS now?

File: next/components/thumbprint-components/props-table/component-definition.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+export interface ComponentDefinition {
+    displayName: string;
+    description: string;
+    methods: unknown[];
```
Comment (lavelle) [inline next/components/thumbprint-components/props-table/component-definition.ts:4]: Is there a reason this is unknown?

File: next/components/thumbprint-components/props-table/component-definition.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    props: Record<
+        string,
+        {
```
Comment (lavelle) [inline next/components/thumbprint-components/props-table/component-definition.ts:7]: Can this be a named type to make the nesting more readable?

File: next/components/mdx/mdx.tsx [hunk 4]
```diff
@@ -289,11 +90,126 @@ export const MDXComponentPage = ({
...
+                                                            elementName="div"
+                                                        >
+                                                            <div className="b black-300">Type</div>
+                                                            <PropType
+                                                                type={prop.tsType.name}
+                                                                value={prop.tsType.elements}
+                                                            />
+                                                        </Text>
+                                                    )}
+                                                    {get(prop.defaultValue, 'value') && (
```
Comment (lavelle) [inline next/components/mdx/mdx.tsx:185]: prefer `?.` to `_.get` in TS files

**Review by lavelle — CHANGES_REQUESTED** [review]

File: next/components/thumbprint-components/props-table/prop-type.jsx (hunk 1 above)
Comment (danoc) [reply next/components/thumbprint-components/props-table/prop-type.jsx:1]: Yeah this one is copied over.

File: next/components/mdx/mdx.tsx (hunk 4 above)
Comment (danoc) [reply next/components/mdx/mdx.tsx:185]: I should've indicated this in the PR description, but this file was almost entirely copied over from the Gatsby code. I'd like to leave this as-is for now.

File: next/components/thumbprint-components/props-table/component-definition.ts (hunk 2 above)
Comment (danoc) [reply next/components/thumbprint-components/props-table/component-definition.ts]: This only ever gets populated by class components, I think. I've updated this PR to expand on this type definition a bit.

File: next/components/thumbprint-components/props-table/component-definition.ts (hunk 3 above)
Comment (danoc) [reply next/components/thumbprint-components/props-table/component-definition.ts]: Done

**Review by lavelle — APPROVED** [review]

**Pull request merged**